test/runtest.py

```diff
@@ -1207,6 +1207,45 @@ def test_multi_threading_in_worker():
         # test multi-threading in the worker
         ray.get(test_multi_threading_in_worker.remote())
 
+    @unittest.skipIf(
+        os.environ.get("RAY_USE_XRAY") != "1",
+        "This test only works with xray.")
+    def testFreeObjects(self):
+        ray.init(num_workers=1, use_raylet=True)
+
+        @ray.remote
+        def f():
+            return 1
+
+        @ray.remote
+        def g():
+            return 2
+
+        def create():
+            a = f.remote()
+            b = g.remote()
+            (l1, l2) = ray.wait([a, b], num_returns=2)
+            assert len(l1) == 2
+            assert len(l2) == 0
+            return (a, b)
+
+        def flush():
+            # Flush the Release History.
+            print("Start Flush!")
+            for i in range(1000):
+                f.remote()
+                g.remote()
+            print("Flush finished!")
+
+        (a, b) = create()
+        ray.internal.free([a, b])
+
+        flush()
+        (l1, l2) = ray.wait([a, b], timeout=1)
+        # The objects are deleted.
+        assert len(l1) == 0
+        assert len(l2) == 2
+
```

Comment:
We should probably also test this with multiple raylets, right? Which tests the regular and the `local_only` code path. What do you think?

Reply:
Is it possible to test with multiple raylets in `test/runtest.py`? Otherwise I need to test it in Jenkins Docker environment.

Reply:
Yes! See, e.g., https://github.com/ray-project/ray/blob/9825da723380f42ab7a24f7784eb5255a154dac8/test/runtest.py#L1673-L1682